lib/fog/azurerm/models/network/virtual_network.rb

```diff
@@ -8,37 +8,70 @@ class VirtualNetwork < Fog::Model
         attribute :id
         attribute :location
         attribute :resource_group
-        attribute :dns_list
-        attribute :subnet_address_list
-        attribute :network_address_list
+        attribute :dns_servers
+        attribute :subnets
+        attribute :address_prefixes
 
         def self.parse(vnet)
           hash = {}
           hash['id'] = vnet['id']
           hash['name'] = vnet['name']
           hash['resource_group'] = vnet['id'].split('/')[4]
           hash['location'] = vnet['location']
-          hash['dns_list'] = vnet['properties']['dhcpOptions']['dnsServers'].join(',') unless vnet['properties']['dhcpOptions'].nil?
-          hash['network_address_list'] = vnet['properties']['addressSpace']['addressPrefixes'].join(',') unless vnet['properties']['addressSpace']['addressPrefixes'].nil?
-          subnet_address_list = []
+          hash['dns_servers'] = vnet['properties']['dhcpOptions']['dnsServers'] unless vnet['properties']['dhcpOptions'].nil?
+          hash['address_prefixes'] = vnet['properties']['addressSpace']['addressPrefixes'] unless vnet['properties']['addressSpace']['addressPrefixes'].nil?
+
+          subnets = []
           vnet['properties']['subnets'].each do |subnet|
-            subnet_address_list << subnet['properties']['addressPrefix']
+            subnet_object = Fog::Network::AzureRM::Subnet.new
+            subnets << subnet_object.merge_attributes(Fog::Network::AzureRM::Subnet.parse(subnet))
           end
-          hash['subnet_address_list'] = subnet_address_list.join(',')
+          hash['subnets'] = subnets
           hash
         end
 
         def save
           requires :name
           requires :location
           requires :resource_group
-          vnet = service.create_virtual_network(resource_group, name, location, dns_list, subnet_address_list, network_address_list)
+          validate_subnets(subnets) unless subnets.nil?
+
+          vnet = service.create_virtual_network(resource_group, name, location, dns_servers, subnets, address_prefixes)
           merge_attributes(Fog::Network::AzureRM::VirtualNetwork.parse(vnet))
         end
 
         def destroy
           service.delete_virtual_network(resource_group, name)
         end
+
+        private
+
+        def validate_subnets(subnets)
+          if subnets.is_a?(Array)
+            subnets.each do |subnet|
+              if subnet.is_a?(Hash)
+                validate_subnet_params(subnet)
+              else
+                raise(ArgumentError, ':subnets must be an Array of Hashes')
+              end
+            end
+          else
+            raise(ArgumentError, ':subnets must be an Array')
+          end
+        end
+
+        def validate_subnet_params(nsr)
+          required_params = [
+            :name,
+            :address_prefix
+          ]
+          missing = required_params.select { |p| p unless nsr.key?(p) }
+          if missing.length == 1
```

Comment:
Implement a utility method, that takes as input required argument list and input argument list. check_for_required_arguments()
It should raise ArgumentError with proper messaging and which argument(s) are missing.